### ISSUE TITLE
Different behavior during Nuxt 3 SSR call

### ISSUE THREAD

**Issue by yasin-mesut** [nuxt/nuxt#25564]
### Environment

- Operating System: Darwin
- Node Version:     v20.5.0
- Nuxt Version:     3.9.3
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  yarn@1.22.21
- Builder:          -
- User Config:      devtools, devServer, nitro, components, css, modules, stylelint, runtimeConfig, sitemap, serverHandlers, vite, hooks
- Runtime Modules:  nuxt-simple-sitemap@4.1.15, @pinia/nuxt@0.5.1, @artmizu/nuxt-prometheus@2.2.0, @vueuse/nuxt@10.7.0, @nuxtjs/stylelint-module@5.1.0
- Build Modules:    -

### Reproduction

-

### Describe the bug

I'm trying to call an api throughout a http proxy, which looks somewhat like this for my own Nuxt 3 api endpoint.

file: `server/api/custom.ts`

```
import axios from 'axios';
import httpsProxyAgent from 'https-proxy-agent'; // running on version 5.0.1

axios.get("https://some-api.com/foo/bar", {
  httpsAgent: httpsProxyAgent({
            host: process.env.PROXY_HOST!,
            port: parseInt(process.env.PROXY_PORT!),
        });
})
```
Accessing this url directly works.

`http://localhost:3000/api/custom`

However, when running within a page and using `useAsyncData` against my own endpoint, I run into a protocol issue.

So this ...

```
await useAsyncData(async () => {
    return (await $fetch('/api/custom'));
});
```

Now results in this ...

`TypeError [ERR_INVALID_PROTOCOL]: Protocol "https:" not supported. Expected "http:"`

This only happens because the proxy is forwarding the requested url through `http` and the third-party api is redirecting to `https`. So naturally, using a `https` proxy would solve the issue. However, that's not possible. On the other hand, accessing the exact same endpoint directly through my custom api (without running a server side rendered page using `useAsyncData`) it works. So it seems like there's a difference between Nuxt 3 api usage as well as the SSR environment.

Hint: This issue just came up recently when we migrated the application from Nuxt 2 to Nuxt 3. So basically, the exact same code was behaving differently and just worked in the past.

Long story short: Is there any possible fix for that?

**Comment by amandesai01** [comment]
with nuxt2, almost all developers used @nuxtjs/axios I hope you did as well, whereas, nuxt3 uses unjs/ofetch for fetching. maybe, this issue is relevant there?

**Comment by madebyfabian** [comment]
@yasin-mesut Do you still experience this issue? If so, please add a working, minimal reproduction
